File: android/src/main/java/com/yugioh/android/base/ISliding.java
package com.yugioh.android.base;

import android.view.View;
import android.view.ViewGroup.LayoutParams;

public interface ISliding {

    void setBehindContentView(View view, LayoutParams layoutParams);

    void setBehindContentView(View view);

    void setBehindContentView(int layoutResID);

    SlidingMenu getSlidingMenu();

    void toggle();

    void showContent();

    void showMenu();

    void showSecondaryMenu();

    void setSlidingActionBarEnabled(boolean slidingActionBarEnabled);

}

File: android/src/main/kotlin/com/yugioh/android/base/ISliding.kt
package com.yugioh.android.base

import android.view.View
import android.view.ViewGroup

/**
 * Created by rarnu on 3/23/16.
 */
interface ISliding {

    fun setBehindContentView(v: View?, params: ViewGroup.LayoutParams?)

    fun setBehindContentView(v: View?)

    fun setBehindContentView(id: Int)

    fun getSlidingMenu(): SlidingMenu?

    fun toggle()

    fun showContent()

    fun showMenu()

    fun showSecondaryMenu()

    fun setSlidingActionBarEnabled(slidingActionBarEnabled: Boolean)
}